python/cudf/cudf/core/series.py

```diff
@@ -4157,7 +4158,14 @@ def hash_encode(self, stop, use_name=False):
         if not stop > 0:
             raise ValueError("stop must be a positive integer.")
 
-        initial_hash = [hash(self.name) & 0xFFFFFFFF] if use_name else None
+        name_hasher = sha256()
```

Comment:
My choice of SHA-256 for the cryptographic hash function is somewhat arbitrary. Any of the hash functions from hashlib would be valid as long as it's FIPS compliant and among those that are "always present in this module": https://docs.python.org/3/library/hashlib.html#hash-algorithms